### PR TITLE
MCR-3028 make MyCoRe compile with Java 21

### PR DESCRIPTION
**Pull request by yagee-de**
update to PMD 7.0.0
fix newly detected PMD violations

[Link to jira](https://mycore.atlassian.net/browse/MCR-3028).

### CODE REVIEW

**Comment by yagee-de** [comment]
TODO: test removal of parallel execution

File: mycore-viewer/src/test/resources/mycore-home/mycore-viewer/mycore.properties [hunk 1]
```diff
@@ -1 +1,10 @@
 MCR.Persistence.Database.Enable=false
+MCR.Resource.Resolver=mets
```
Comment (toKrause) [inline mycore-viewer/src/test/resources/mycore-home/mycore-viewer/mycore.properties]: Copy-Paste from `mycore-mets/src/test/resources/mycore-home/mycore-mets/mycore.properties`? Name should probably be `viewer` here, or, better, `test`in both cases, to make future copy-pasting simpler.

File: mycore-classeditor/src/main/java/org/mycore/frontend/classeditor/json/MCRJSONCategoriesSaveList.java [hunk 2]
```diff
@@ -1,59 +0,0 @@
...
-
-import java.util.ArrayList;
-import java.util.Objects;
-
-import org.mycore.datamodel.classifications2.MCRCategory;
-import org.mycore.datamodel.classifications2.MCRCategoryID;
-
-import jakarta.ws.rs.WebApplicationException;
-
-public class MCRJSONCategoriesSaveList {
```
Comment (toKrause) [inline mycore-classeditor/src/main/java/org/mycore/frontend/classeditor/json/MCRJSONCategoriesSaveList.java:29]: Why is this class deleted?

**Review by toKrause — COMMENTED** [review]
I'm a bit unsure about this PR.

The majority of the changes seem to be completely unrelated to the task at hand.

As per the "scout rule", I have no problem with the small cleanup refactorings (like deleting the unused `tearDown` method or removing unnecessary explicit class names when calling static methods), but, for example, the redesign of the initialization and now also re-initialization of `MCRURIResolver` or the seemingly unprovoked deletion of `MCRJSONCategoriesSaveList` appear to be very out-of-scope.

I think "Compile MyCoRe with Java 21" and "Update to PMD 7.0.0 and fix newly detected PMD violations" should be separate PRs.

**Comment by yagee-de** [comment]
> I'm a bit unsure about this PR.
> 
> The majority of the changes seem to be completely unrelated to the task at hand.
> 
> As per the "scout rule", I have no problem with the small cleanup refactorings (like deleting the unused `tearDown` method or removing unnecessary explicit class names when calling static methods), but, for example, the redesign of the initialization and now also re-initialization of `MCRURIResolver` or the seemingly unprovoked deletion of `MCRJSONCategoriesSaveList` appear to be very out-of-scope.
> 
> I think "Compile MyCoRe with Java 21" and "Update to PMD 7.0.0 and fix newly detected PMD violations" should be separate PRs.

Every change is necessary, so is the PMD update. The old version does not run with Java 21 compiled classes. The deleted class for example contains PMD issues and is not used anywhere throughout the code. So deleting was the easiest fix. MCRURIResolver.reInit() is due to tests to work independent of other tests that were running in the same JVM. That cut the execution time for compiling and testing mycore-base from 7:30 min do about 2 min on my computer. Saves nerves!

File: mycore-viewer/src/test/resources/mycore-home/mycore-viewer/mycore.properties (hunk 1 above)
Comment (yagee-de) [reply mycore-viewer/src/test/resources/mycore-home/mycore-viewer/mycore.properties]: applied suggested changes

File: mycore-classeditor/src/main/java/org/mycore/frontend/classeditor/json/MCRJSONCategoriesSaveList.java (hunk 2 above)
Comment (yagee-de) [reply mycore-classeditor/src/main/java/org/mycore/frontend/classeditor/json/MCRJSONCategoriesSaveList.java:29]: did contain PMD issues and was not used throughout the code

**Review by Mewel — DISMISSED** [review]

**Comment by yagee-de** [comment]
Build fails occasionally
https://github.com/MyCoRe-Org/mycore/actions/runs/8168602587/job/22349054496?pr=2049#step:8:2453
with
`Error:  Failed to execute goal org.apache.maven.plugins:maven-pmd-plugin:3.21.2:pmd (default) on project mycore-base: Execution default of goal org.apache.maven.plugins:maven-pmd-plugin:3.21.2:pmd failed: An API incompatibility was encountered while executing org.apache.maven.plugins:maven-pmd-plugin:3.21.2:pmd: java.lang.NoSuchMethodError: 'java.lang.String net.sourceforge.pmd.Report$ProcessingError.getFile()'`

This should be fixed https://github.com/pmd/pmd/commit/f73d75d5a363f9c09f3880948944a2028b52a6f0 in a future release of PMD.

**Review by rsteph-de — APPROVED** [review]
looks good for me

**Pull request merged**